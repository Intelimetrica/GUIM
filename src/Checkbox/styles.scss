@import "../styles/colors.scss";
@import "../styles/themes.scss";

.GUIMCheckbox {
  -webkit-appearance: none;
  background-color: var(--white);
  border: 2px solid var(--main-gray);
  border-radius: 3px;
  display: inline-block;
  position: relative;
  outline: none;
  width: 16px;
  height: 16px;
  transition: all 0.5s ease;
  cursor: pointer;
  &:checked {
    &::before {
      -webkit-transform: rotate(45deg);
      transform: rotate(45deg);
      content: "";
      position: absolute;
      top: 0px;
      left: 5px;
      display: table;
      width: 2px;
      height: 10px;
      border: 2px solid #fff;
      border-top-width: 0;
      border-left-width: 0;
    }
  }
  &:disabled {
    cursor: default;
    opacity: 0.4;
  }
}
.GUIMSwitch {
  display: flex;
  width: 50px;
  height: 22px;
  &.Disabled > * {
    opacity: 0.4;
    cursor: not-allowed !important;
  }
}

.GUIMSwitch label {
  cursor: pointer;
  text-indent: -9999px;
  width: 40px;
  height: 22px;
  background: #bbbbbb;
  display: block;
  border-radius: 90px;
  position: relative;
}
.GUIMSwitch input[type="checkbox"] {
  height: 0;
  width: 0;
  visibility: hidden;
}
.GUIMSwitch label:after {
  content: "";
  position: absolute;
  top: 1px;
  left: 1px;
  width: 18px;
  height: 18px;
  background: #fff;
  border-radius: 90px;
  transition: 0.1s;
}
.GUIMSwitch input:not(:checked) + label:after {
  top: 2px;
  left: 2px;
}
.GUIMSwitch.Checked label {
  height: 20px;
  width: 38px;
}
.GUIMSwitch input:checked + label:after {
  left: calc(100% - 1px);
  transform: translateX(-100%);
}
.GUIMSwitch label:active:after {
  width: 18px;
}

.GUIMCheckboxBlue:checked {
  @apply --blue-theme;
}
.GUIMCheckboxGray:checked {
  @apply --gray-theme;
}
.GUIMCheckboxGreen:checked {
  @apply --green-theme;
}
.GUIMCheckboxOrange:checked {
  @apply --orange-theme;
}
.GUIMSwitchBlue:checked + label {
  @apply --blue-theme;
}
.GUIMSwitchGray:checked + label {
  @apply --gray-theme;
}
.GUIMSwitchGreen:checked + label {
  @apply --green-theme;
}
.GUIMSwitchOrange:checked + label {
  @apply --orange-theme;
}
