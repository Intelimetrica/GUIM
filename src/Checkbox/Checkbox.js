import React from "react";
import "./styles.scss";

const Checkbox = props => ( props.isSwitch ? (
  <div className={props.isSwitch ? 'GUIMSwitch' : null}>
    <input
      id={`switch${props.id}`}
      className={`GUIMCheckbox  ${props.className} ${themes[props.color] || themes["blue"]}`}
      style={props.styles}
      type="checkbox"
      name={props.name}
      onChange={() => props.disabled ? null : props.onChange()}
      checked={props.checked}
      disabled={props.disabled}
    />
     <label htmlFor={`switch${props.id}`}>.</label>
      
  </div>
) : (
  <input
      id={`switch${props.id}`}
      className={`GUIMCheckbox  ${props.className} ${themes[props.color] || themes["blue"]}`}
      style={props.styles}
      type="checkbox"
      name={props.name}
      onChange={() => props.disabled ? null : props.onChange()}
      checked={props.checked}
      disabled={props.disabled}
    />
)
);

Checkbox.defaultProps = {
  name: "checkbox",
  onChange: () => console.log("Checkbox onChange()"),
  checked: false,
  styles: {},
  className: "",
  color: "blue",
  guimInput: "checkbox",
  disabled: false,
  isSwitch: false,
  id: Math.random() * 100
};

const themes = {
  blue: "GUIMCheckboxBlue",
  gray: "GUIMCheckboxGray",
  green: "GUIMCheckboxGreen",
  orange: "GUIMCheckboxOrange"
};

export default Checkbox;
