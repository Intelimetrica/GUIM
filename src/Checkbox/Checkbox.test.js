import React from 'react';
import Checkbox from './Checkbox';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('<Checkbox />', () => {
  describe('behaviour tests', () => {
    let checked = false;
    let component;
    let tree;
    let reRender;
    const makeRenderer = component =>
      () => tree = component.toJSON();

    beforeEach(() => {
      component = renderer.create(
        <Checkbox
          onChange={_e => checked = !checked}
          checked={checked}
        />);
      reRender = makeRenderer(component);
      reRender();
    })

    it('match snapshot', () => {
      expect(tree).toMatchSnapshot();
    });


    it('is true after a click', () => {
      tree.props.onChange();
      reRender();
      expect(checked).toBe(true);
    });

    it('is false after two clicks', () => {
      tree.props.onChange();
      tree.props.onChange();
      reRender();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('disabled behaviour tests', () => {
    let checked = false;
    let component;
    let tree;
    let reRender;
    const makeRenderer = component =>
      () => tree = component.toJSON();

    beforeEach(() => {
      component = renderer.create(
        <Checkbox
          onChange={_e => checked = !checked}
          checked={checked}
          disabled
        />);
      reRender = makeRenderer(component);
      reRender();
    })

    it('match snapshot', () => {
      expect(tree).toMatchSnapshot();
    });


    it('still false after a click', () => {
      tree.props.onChange();
      reRender();
      expect(checked).toBe(false);
    });

  });

  describe('dom tests', () => {
    let checked = false;
    let component;

    beforeEach(() => {
      component = shallow(
        <Checkbox
          name='test_checkbox'
          onChange={_e => checked = !checked}
          checked={checked}
        />
      );
    })

    it('renders an input', () => {
      expect(component.name()).toBe("input");
    });

    it('name is test_checkbox', () => {
      expect(component.prop('name')).toBe("test_checkbox");
    });
  });

});
