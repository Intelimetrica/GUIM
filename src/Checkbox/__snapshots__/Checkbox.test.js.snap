// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> behaviour tests is false after two clicks 1`] = `
<div
  className="GUIMSwitch Checked"
>
  <input
    checked={true}
    className="Switch GUIMSwitchBlue"
    disabled={false}
    id="switch-00-0"
    name="checkbox"
    onChange={[Function]}
    style={Object {}}
    type="checkbox"
  />
  <label
    htmlFor="switch-00-0"
  >
    .
  </label>
</div>
`;

exports[`<Checkbox /> behaviour tests match snapshot 1`] = `
<div
  className="GUIMSwitch "
>
  <input
    checked={false}
    className="Switch GUIMSwitchBlue"
    disabled={false}
    id="switch-00-0"
    name="checkbox"
    onChange={[Function]}
    style={Object {}}
    type="checkbox"
  />
  <label
    htmlFor="switch-00-0"
  >
    .
  </label>
</div>
`;

exports[`<Checkbox /> disabled behaviour tests match snapshot 1`] = `
<input
  checked={false}
  className="GUIMCheckbox  GUIMCheckboxBlue"
  disabled={true}
  id="switch-00-1"
  name="checkbox"
  onChange={[Function]}
  style={Object {}}
  type="checkbox"
/>
`;
